### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by serena-ruan**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Update PR template 'docs update' section.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR need docs update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] References
  - [ ] Instructions

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -23,10 +23,14 @@ If you're unsure about what to test, where to add tests, or how to run tests, pl
...
-## Does this PR change the documentation?
+## Does this PR need docs update?
 
 - [ ] No. You can skip the rest of this section.
-- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.
+- [ ] Yes. Make sure you have verified below sections.
+  - [ ] Any existing samples / documentation have been updated.
+  - [ ] Any additional relevant samples / documentation has been added.
+  - [ ] Any docs reference has been checked / updated.
+  - [ ] Changed pages / sections render correctly in the documentation preview.
```
Comment (harupy) [inline .github/pull_request_template.md:33]: Can we simplify this? People don't read long text.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -23,10 +23,14 @@ If you're unsure about what to test, where to add tests, or how to run tests, pl
...
-## Does this PR change the documentation?
+## Does this PR need docs update?
 
 - [ ] No. You can skip the rest of this section.
-- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.
+- [ ] Yes. Make sure you have verified below sections.
+  - [ ] Existing samples / documentation have been updated.
+  - [ ] Additional relevant samples / documentation have been added.
+  - [ ] Docs references have been checked / updated.
+  - [ ] Changed pages / sections render correctly in the documentation preview.
```
Comment (serena-ruan) [inline .github/pull_request_template.md:33]: /ai Can you simplify this?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -26,10 +26,10 @@ Please describe how you confirmed the proposed feature/bug-fix/change works here
...
 - [ ] No. You can skip the rest of this section.
-- [ ] Yes. Make sure you have verified below sections.
-  - [ ] Existing samples / documentation have been updated.
-  - [ ] Additional relevant samples / documentation have been added.
-  - [ ] Docs references have been checked / updated.
+- [ ] Yes. Please verify the following:
+  - [ ] Update existing samples / documentation.
+  - [ ] Add relevant samples / documentation.
+  - [ ] Check / update docs references.
   - [ ] Changed pages / sections render correctly in the documentation preview.
```
Comment (harupy) [inline .github/pull_request_template.md:33]: What is the difference between the first three items?

Comment (serena-ruan) [reply .github/pull_request_template.md:33]: 1 refers to existing samples, if you just change the function usage. 2 refers to new samples/documentation if it's a new feature. Maybe 3 could be removed, I mean to check the docs links.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -23,10 +23,14 @@ If you're unsure about what to test, where to add tests, or how to run tests, pl
...
-## Does this PR change the documentation?
+## Does this PR need docs update?
 
 - [ ] No. You can skip the rest of this section.
-- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.
+- [ ] Yes. Please verify the following:
+  - [ ] Update existing samples / documentation.
+  - [ ] Add relevant samples / documentation.
+  - [ ] Check / update docs references.
+  - [ ] Changed pages / sections render correctly in the documentation preview.
```
[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: /ai merge these items into a single sentence. Make sure the sentence is short.

[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: /ai merge these items into a single sentence. Make sure the sentence starts with "I've verified".

[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: Can we just say "Yes, I've updated examples, docs, references."

[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: We can remove `Changed pages / sections render correctly in the documentation preview.`. We have a separate message for this.

[the remark this answers is not in the document]
Comment (serena-ruan) [reply .github/pull_request_template.md:33]: > Can we just say "Yes, I've updated docs, examples, and references."

I think only when people check those check boxes, they verify they have done the checks. One single line seems less constraint.

[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: Got it, then can we do the following?

```suggestion
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] References
  - [ ] Instructions
```

[the remark this answers is not in the document]
Comment (serena-ruan) [reply .github/pull_request_template.md:33]: Do you want to keep the `Changed pages / sections render correctly in the documentation preview.` in the original section of `## Does this PR change the documentation?`

[the remark this answers is not in the document]
Comment (harupy) [reply .github/pull_request_template.md:33]: No, because we have this: https://github.com/mlflow/mlflow/pull/9397#issuecomment-1685768115

**Review by WeichenXu123 — APPROVED** [review]

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -23,10 +23,13 @@ If you're unsure about what to test, where to add tests, or how to run tests, pl
...
 -->
 
-## Does this PR change the documentation?
+## Does this PR need docs update?
 
 - [ ] No. You can skip the rest of this section.
-- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.
+- [ ] Yes. I've updated:
+  - [ ] Examples
+  - [ ] References
```
Comment (harupy) [inline .github/pull_request_template.md:31]:
```suggestion
  - [ ] API references
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -23,10 +23,13 @@ If you're unsure about what to test, where to add tests, or how to run tests, pl
 Please describe how you confirmed the proposed feature/bug-fix/change works here. For example, if you fixed an MLflow client API, you could attach the code that didn't work prior to the fix but works now, or if you added a new feature on MLflow UI, you could attach a video that demonstrates the feature.
 -->
 
-## Does this PR change the documentation?
+## Does this PR need docs update?
```
Comment (harupy) [inline .github/pull_request_template.md:26]: /ai is "docs update" grammatically correct?

**Pull request merged**